implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -314,7 +329,33 @@ boolean isAddDiscoveredInterceptors() {
 
     @Override
     public SmallRyeConfig build() {
-        return new SmallRyeConfig(this);
+        ConfigMappingProvider mappingProvider = mappingsBuilder.build();
+        if (!defaultValues.isEmpty() || !mappingProvider.getDefaultValues().isEmpty()) {
+            final KeyMap<String> mappingProviderDefaultValues = mappingProvider.getDefaultValues();
+            defaultValues.forEach((key, value) -> mappingProviderDefaultValues.findOrAdd(key).putRootValue(value));
+            withSources(new KeyMapBackedConfigSource("DefaultValuesConfigSource", mappingProviderDefaultValues) {
```

Comment:
This no longer needs to be an anonymous class AFAICT; the ordinal can be passed in, and the properties map should already be empty.

Reply:
Sort of. `KeyMapBackedConfigSource` is abstract. Do we want to make an implementation for this case?

Reply:
It probably can be made concrete then, I would think.  I don't imagine any possible subclass might do more than what the base class already does.

Reply:
Ok!